File: vigenere.c
// Vigenere cipher
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

// Declares prototype for code shift, defining at the end
int shift(char c);

int main(int argc, string argv[])
{
    // Checks if there are two arguments or strings
    if (argc == 2)
    {
        // Iterates for each character in argv 
        for (int i = 0; i < strlen(argv[1]); i++)
        {
            // Checks if every character in argv is alphanumeric
            if (isalpha(argv[1][i]))
            {
            }
            else
            {
                printf("Usage: ./vigenere keyword\n");
                return 1;
            }
        }
        // Prompting for plaintext
        string pt = get_string("plaintext: ");
        printf("ciphertext: ");
        // Iterates each character in plaintext
        for (int k = 0; k < strlen(pt); k++)
        {
            int j = k;
            while (j >= strlen(argv[1]))
            {
                j = j - strlen(argv[1]);
            }
            // Ciphers uppercase letters
            if (pt[k] >= 'A' && pt[k] <= 'Z')
            {
                // Protection against going out of letters
                if (pt[k] + shift(argv[1][j]) > 'Z')
                {
                    printf("%c", pt[k] - 26 + shift(argv[1][j]));
                }
                else
                {
                    printf("%c", pt[k] + shift(argv[1][j]));
                }
            }
            // Ciphers lowercase letters
            else
            {
                if (pt[k] >= 'a' && pt[k] <= 'z')
                {
                    // Again, protection against going out of letters
                    if (pt[k] + shift(argv[1][j]) > 'z')
                    {
                        printf("%c", pt[k] - 26 + shift(argv[1][j]));
                    }
                    else
                    {
                        printf("%c", pt[k] + shift(argv[1][j]));
                    }
                }
                // Prints characters that aren't alphabetical
                else
                {
                    printf("%c", pt[k]);
                }
            }
        }
        printf("\n");
        return 0;
    }
    else
    {
        printf("Usage: ./vigenere keyword\n");
        return 1;
    }
}

// Implements shift: A or a = 0, B or b = 1 etc.
int shift(char c)
{
    if (isupper(c))
    {
        return c - 'A';
    }
    else
    {
        return c - 'a';
    }
}  

File: vigenere.py
# import get string and sys
from cs50 import get_string
import sys

# check if there are two arguments (strings) and argv[1] is alphabetical
if len(sys.argv) != 2:
    print("Usage: 'python vigenere.py key', where the key is a word")
    sys.exit([1])
key = sys.argv[1]
for k in key:
    if k.isalpha() == False:
        print("only alphabetical characters")
        sys.exit([2])

# get plaintext to cipher
p = get_string("plaintext: ")
print("ciphertext: ", end="")

length = len(key)
count = 0

# iterate each character in plaintext
for c in p:

    # cipher only for alphabetic characters
    if c.isalpha() == True:

        # get back to beginning of the key word if already ended
        while count >= length:
            count = count - length

        # create integer to shift by position of character in alphabet
        if key[count].isupper() == True:
            shift = ord(sys.argv[1][count]) - 65
        else:
            shift = ord(sys.argv[1][count]) - 97

        # shift c by key, substracting difference between ascii value and position in alphabet
        if c.isupper() == True:
            c = ((ord(c) - 65 + shift) % 26) + 65
        else:
            c = ((ord(c) - 97 + shift) % 26) + 97

        # change back c to char, print it and count+1
        c = chr(c)
        print(c, end="")
        count += 1
    else:
        print(c, end="")
print()

